// @smartsupp/smartsupp-message@0.13.1/lib/message-content.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.MessageContentType = exports.MessageContent = void 0;
var MessageContent;
(function(MessageContent) {
    let Message;
    (function(Message) {
        Message.Type = {
            Text: 'text',
            Upload: 'upload',
            RateForm: 'rate_form',
            TicketForm: 'ticket_form',
        };
    })(Message = MessageContent.Message || (MessageContent.Message = {}));
    let Event;
    (function(Event) {
        Event.Type = {
            AgentJoin: 'agent_join',
            AgentAssign: 'agent_assign',
            AgentUnassign: 'agent_unassign',
            AgentLeave: 'agent_leave',
            ChatClose: 'chat_close',
            ChatVisitorClose: 'chat_visitor_close',
        };
    })(Event = MessageContent.Event || (MessageContent.Event = {}));
    let Help;
    (function(Help) {
        Help.Type = {
            Text: 'text',
            FoundEmail: 'found_email',
        };
    })(Help = MessageContent.Help || (MessageContent.Help = {}));
})(MessageContent = exports.MessageContent || (exports.MessageContent = {}));
exports.MessageContentType = {
    ...MessageContent.Message.Type,
    ...MessageContent.Event.Type,
    ...MessageContent.Help.Type,
};
